src/Konami.tsx

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+
+export interface KonamiProps {
+  className?: string;
+  code?: number[];
+  disabled?: boolean;
+  resetDelay?: number;
+  action?: () => void;
+  timeout?: number;
+  onTimeout?: () => void;
+}
+
+export interface KonamiStates {
+  done: boolean;
+  input: number[];
+}
+
+class Timer {
+  t: number;
+  timerObj: number | null;
+  fn: TimerHandler;
+
+  constructor(fn: () => void, t: number) {
+    this.t = t;
+    this.fn = fn;
+
+    this.timerObj = setInterval(this.fn, this.t);
+  }
+
+  stop() {
+    if (this.timerObj) {
+      clearInterval(this.timerObj);
+      this.timerObj = null;
+    }
+    return this;
+  }
+
+  start() {
+    if (!this.timerObj) {
+      this.stop();
+      this.timerObj = setInterval(this.fn, this.t);
+    }
+    return this;
+  }
+
+  reset(newT: number) {
+    this.t = newT;
+    return this.stop().start();
+  }
+}
+
+class Konami extends React.Component<KonamiProps, KonamiStates> {
+  private timeoutFunc: ReturnType<typeof setTimeout> | null | undefined;
+  private _timer: any;
+
+  static defaultProps = {
+    action: null,
+    className: '',
+    code: [38, 38, 40, 40, 37, 39, 37, 39, 66, 65],
+    disabled: false,
+    onTimeout: null,
+    resetDelay: 1000,
+    timeout: null,
+  };
+
+  constructor(props: KonamiProps) {
+    super(props);
+
+    this.state = {
+      done: false,
+      input: [],
+    };
+
+    this.timeoutFunc = null;
+    this.onKeyUp = this.onKeyUp.bind(this);
+    this.resetInput = this.resetInput.bind(this);
+  }
+
+  componentDidMount() {
+    const { resetDelay } = this.props;
+
+    document.addEventListener('keyup', this.onKeyUp);
+    const delay = Number(resetDelay);
+    if (delay !== 0) {
+      this._timer = new Timer(() => this.resetInput(), delay);
+    }
+  }
+
+  componentWillUnmount() {
+    const { resetDelay } = this.props;
+
+    clearTimeout(Number(this.timeoutFunc ?? undefined));
```

Comment:
Why are you casting the `clearTimeout` argument to `Number`?

Reply:
I made that because declared `timeoutFunc` (now `timeoutID`) as `ReturnType` of `setTimeout`, resulting in `Timeout` type, and `clearTimeout` expects a number.

I will modify the `timeoutFunc` type to `number` to avoid that parsing

Reply:
I made that because in line 128 the `setTimeout` instance is passed to `timeoutFunc`. The return of `setTimeout` it's `NodeJS.Timeout` type.

https://github.com/vmarchesin/react-konami-code/blob/f8964542e1275e1fea1e3c9f9a89905f8f1af755/src/Konami.tsx#L128

So to access the ID of `Timeout` it's needed to parse that value to `number`

Reply:
That's weird, since the behavior for `setInterval` and `clearInterval` is exactly the same and there's no need to parse the return to `Number` in that case (for `timerIntervalID`).

Reply:
Sorry, that was my bad.
It's true that isn't needed to parse the `timeoutID`.

I solve with a simple condition:

```typescript
if (this.timeoutID) clearTimeout(this.timeoutID);
```